src/uu/top/src/top.rs

```diff
@@ -0,0 +1,239 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use std::{thread::sleep, time::Duration};
+
+use clap::{arg, crate_version, ArgAction, ArgGroup, ArgMatches, Command};
+use picker::sysinfo;
+use prettytable::{format::consts::FORMAT_CLEAN, Row, Table};
+use sysinfo::{Pid, Users};
+use uucore::{
+    error::{UResult, USimpleError},
+    format_usage, help_about, help_usage,
+};
+
+const ABOUT: &str = help_about!("top.md");
+const USAGE: &str = help_usage!("top.md");
+
+mod field;
+mod picker;
+
+#[allow(unused)]
+#[derive(Debug)]
+enum Filter {
+    Pid(Vec<u32>),
+    User(String),
+    EUser(String),
+}
+
+#[derive(Debug)]
+struct Settings {
+    // batch:bool
+    filter: Option<Filter>,
+    width: Option<usize>,
+}
+
+impl Settings {
+    fn new(matches: &ArgMatches) -> Self {
+        let filter = if let Some(pidlist) = matches.get_many::<u32>("pid") {
+            Some(Filter::Pid(pidlist.cloned().collect::<Vec<_>>()))
+        } else if let Some(user) = matches.get_one::<String>("filter-any-user") {
+            Some(Filter::User(user.clone()))
+        } else {
+            matches
+                .get_one::<String>("filter-only-euser")
+                .map(|euser| Filter::EUser(euser.clone()))
+        };
+
+        let width = matches.get_one::<usize>("width").cloned();
+
+        Self { filter, width }
+    }
+}
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    // Must refresh twice.
+    picker::sysinfo().write().unwrap().refresh_all();
+    sleep(Duration::from_millis(200));
+    picker::sysinfo().write().unwrap().refresh_all();
+
+    let settings = Settings::new(&matches);
+
+    if let Some(Filter::User(user)) = &settings.filter {
+        if Users::new_with_refreshed_list()
+            .iter()
+            .all(|it| it.name() != user)
+        {
+            return Err(USimpleError::new(1, "top: Invalid user"));
```

Comment:
`USimpleError` automatically prefixes the error message with the tool name, so currently the error message shown is "top: top: Invalid user".
```suggestion
            return Err(USimpleError::new(1, "Invalid user"));
```